### PR TITLE
[stable/external-dns] Enable environment variables via "valueFrom"

### LINKED ISSUE #5509: [stable/external-dns] Use preexisting secrets
**Is this a request for help?**:
no
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
Feature request

External-dns currently takes e.g. AWS credentials as values and adds them into a secret. But helm values might be stored in version control and one usually does not want credentials checked into git. It should be possible to use already existing secrets instead. 

**Version of Helm and Kubernetes**:
helm 2.8.2
kubernetes 1.9.5

**Which chart**:
stable/external-dns

### PR DESCRIPTION
**Pull request by simontesar**
**What this PR does / why we need it**: It makes external-dns be able to retrieve credentials from already existing secrets, avoiding duplication and more importantly credentials stored in helm overrides which might be checked into version control.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #5509 

**Special notes for your reviewer**:
Environment variables specified without `valueFrom` still get stored in a secret, as before.

### CODE REVIEW

File: stable/external-dns/templates/deployment.yaml
Comment (unguiculus) [inline stable/external-dns/templates/deployment.yaml:110]: Why not make it simpler and more flexible like so:

```yaml
{{ tpl .Values.extraEnv . | indent 10 }}
```

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by razor-x** [comment]
Anything I can contribute to help push this along?

**Review by unguiculus — APPROVED** [review]

**Comment by razor-x** [comment]
@simt2 Are you still active for this PR? Otherwise I am happen to take ownership to get this merged quickly.

File: stable/external-dns/templates/deployment.yaml
Comment (simontesar) [reply stable/external-dns/templates/deployment.yaml:110]: Because the chart stores environment variables in a secret and I didn't want to break that, although I don't know why it does that in the first place. See `secret.yaml`. I'd also be happy to remove that, if necessary.

**Comment by simt2** [comment]
Sorry for the inactivity, trying to get that merged in the next few days.

**Comment by mattfarina** [comment]
/check-dco

The DCO labeler should only catch one commit missing the signoff because the other is a merge commit. DCO labeler was just fixed so rechecking.

**Comment by razor-x** [comment]
Can someone explain what is blocking this from being merged? I'm happy to do whatever is needed to get this done.

**Comment by razor-x** [comment]
@simt2 Has been inactive for 2 months, hopefully all is ok and they are just no longer available to give the final DCO. 

@unguiculus If that is the case, how can we get this merged without the DCO? Otherwise this may just sit here indefinitely.

**Comment by unguiculus** [comment]
@razor-x Feel free to open a new PR.

**Pull request closed without merging**
